allspice/utils/bom_generation.py

```diff
@@ -231,8 +253,21 @@ def _extract_repetitions(sheet_refs: list[dict]) -> dict[str, int]:
 
     repetitions = {}
     for sheet_ref in sheet_refs:
-        sheet_name = sheet_ref["sheet_name"]["name"]
-        sheet_file_name = sheet_ref["filename"]
+        try:
+            sheet_name = (
+                sheet_ref["sheet_name"]["name"] if sheet_ref["sheet_name"] is not None else ""
+            )
+        except Exception as e:
+            raise ValueError(f"Could not find sheet name in {sheet_ref=}\nCaused by: {e}")
```

Comment:
This could be from `sheet_name` or `name` not existing.

Reply:
While thinking about this I realized we don't need to raise an error here, so I replaced this with `get()`.